guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/RangeDeserializer.java

```diff
@@ -193,6 +216,42 @@ public Range<?> deserialize(JsonParser p, DeserializationContext context)
         }
     }
 
+    private Range<?> deserializeRangeFromString(DeserializationContext context, JsonParser p)
+            throws IOException
+    {
+        String rangeInterval = p.getText();
+
+        if (rangeInterval == null || rangeInterval.isEmpty())
+        {
+            return null;
+        }
+
+        BoundType lowerBoundType = rangeInterval.startsWith("[") ? BoundType.CLOSED : BoundType.OPEN;
```

Comment:
Shouldn't this verify first and last characters to be either `[`or `(` and `]`,or `)`? (as is, could be any old characters)